core-plugins/src/main/java/io/cdap/plugin/batch/source/ExcelInputReader.java

```diff
@@ -309,7 +333,37 @@ public void prepareRun(BatchSourceContext batchSourceContext) throws Exception {
 
   }
 
-  private void createDatasets(@Nullable PipelineConfigurer pipelineConfigurer, @Nullable BatchSourceContext context) {
+  private void createDatasets(PipelineConfigurer pipelineConfigurer) {
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    try {
+      if (!excelInputreaderConfig.containsMacro("errorDatasetName") &&
+        !Strings.isNullOrEmpty(excelInputreaderConfig.errorDatasetName)) {
+        Map<String, String> properties = new HashMap<>();
+        properties.put(Properties.Table.PROPERTY_SCHEMA, errorRecordSchema.toString());
+        properties.put(Properties.Table.PROPERTY_SCHEMA_ROW_FIELD, KEY);
+        DatasetProperties datasetProperties = DatasetProperties.builder().addAll(properties).build();
+
+        pipelineConfigurer.createDataset(excelInputreaderConfig.errorDatasetName, Table.class, datasetProperties);
+      } else if (!excelInputreaderConfig.containsMacro("ifErrorRecord") &&
+        excelInputreaderConfig.ifErrorRecord.equalsIgnoreCase(WRITE_ERROR_DATASET)) {
+        collector.addFailure("Error dataset name should not be empty if choosing write to error "
+                               + "dataset for 'On Error' input.", null)
+          .withConfigProperty(IF_ERROR_RECORD).withConfigProperty(ERROR_DATASET_NAME);
+      }
+
+      if (!excelInputreaderConfig.containsMacro("memoryTableName") &&
+        !Strings.isNullOrEmpty(excelInputreaderConfig.memoryTableName)) {
+        pipelineConfigurer.createDataset(excelInputreaderConfig.memoryTableName, KeyValueTable.class);
+      }
+    } catch (Exception e) {
```

Comment:
don't think anything throws an exception anymore. If not, you should remove this

Reply:
Fixed